Java/core/src/main/java/com/amazon/randomcutforest/sampler/CompactSampler.java

```diff
@@ -338,42 +326,34 @@ public void addPoint(Integer pointIndex) {
     }
 
     private Stream<Weighted<Integer>> streamSample() {
+        reset_weights();
         return IntStream.range(0, size).mapToObj(i -> {
             long index = sequenceIndex != null ? sequenceIndex[i] : SEQUENCE_INDEX_NA;
             return new Weighted<>(pointIndex[i], weight[i], index);
         });
     }
 
     /**
-     * @return the point evicted by the most recent call to {@link #update}, or null
-     *         if no point was evicted.
+     * removes the adjustments to weight in accumulated lambda and resets the
+     * updates to lambda
      */
-    public Optional<ISampled<Integer>> getEvictedPoint() {
-        return Optional.ofNullable(evictedPoint);
+    private void reset_weights() {
```

Comment:
If you reset weights, shouldn't you also set the lambda update time to max sequence index?

Reply:
That would face a check in update() and then move the "clock" to maxSequence index. It is possible that we want that behavior, but maybe have a more focused PR on just that change (along with potentially out of order updates).